ddtrace/sampler.py

```diff
@@ -24,6 +25,7 @@
 from .internal.compat import pattern_type
 from .internal.logger import get_logger
 from .internal.rate_limiter import RateLimiter
+from .internal.utils.cache import cachedmethod
```

Comment:
aside: the `cache` util module could be cythonised for even more performance?

Reply:
yes, that could be a good follow-up action item. now that we have the benchmark case, could be easy to validate (at least in the context of SamplingRule)

Reply:
With only doing `cache.py` -> `cache.pyx`

```
+------------------------------+---------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+
| Benchmark                    | /artifacts/b615b5ec-01ed-4d07-a939-e325c65a9373/sampling_rule_matches/0.56.0rc4.dev20+gaac21153//results.json | /artifacts/b615b5ec-01ed-4d07-a939-e325c65a9373/sampling_rule_matches/0.56.0rc4.dev21+g2a6f8493//results.json |
+==============================+===============================================================================================================+===============================================================================================================+
| samplingrules-high_match     | 58.8 us                                                                                                       | 43.4 us: 1.36x faster                                                                                         |
+------------------------------+---------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+
| samplingrules-average_match  | 59.7 us                                                                                                       | 44.4 us: 1.35x faster                                                                                         |
+------------------------------+---------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+
| samplingrules-low_match      | 65.9 us                                                                                                       | 44.6 us: 1.48x faster                                                                                         |
+------------------------------+---------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+
| samplingrules-very_low_match | 2.38 ms                                                                                                       | 2.03 ms: 1.17x faster                                                                                         |
+------------------------------+---------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+
| Geometric mean               | (ref)                                                                                                         | 1.33x faster                                                                                                  |
+------------------------------+---------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+
```

But obviously we can improve the performance a ton by getting rid of Python data structures.

Probably not worth adding the overhead of managing something else in Cython right now, unless we get more definitive information that this is a bottleneck in creating traces.